fbsimctl/FBSimulatorControlKit/Sources/EventReporter.swift

```diff
@@ -0,0 +1,214 @@
+            return "unknown-name"
+          }
+          return configuration.deviceName
+        case .OSVersion:
+          guard let configuration = simulator.configuration else {
+            return "unknown-os"
+          }
+          return configuration.osVersionString
+        case .State:
+          return simulator.stateString
+        case .ProcessIdentifier:
+          guard let process = simulator.launchdSimProcess else {
+            return "no-process"
+          }
+          return process.processIdentifier.description
+        }
+      }
+      return tokens.joinWithSeparator(" ")
+    }
+  }
+}
+
+public class JSONEventReporter : NSObject, EventReporter {
+  unowned let simulator: FBSimulator
+  let writer: Writer
+  let json: JSON
+
+  init(simulator: FBSimulator, writer: Writer, pretty: Bool) {
+    self.simulator = simulator
+    self.writer = writer
+    self.json = JSON(pretty: pretty)
+    super.init()
+    self.simulator.userEventSink = self
+  }
+
+  public func containerApplicationDidLaunch(applicationProcess: FBProcessInfo!) {
+    self.simulatorEvent()
+  }
+
+  public func containerApplicationDidTerminate(applicationProcess: FBProcessInfo!, expected: Bool) {
+    self.simulatorEvent()
+  }
+
+  public func simulatorDidLaunch(launchdSimProcess: FBProcessInfo!) {
+    self.simulatorEvent()
+  }
+
+  public func simulatorDidTerminate(launchdSimProcess: FBProcessInfo!, expected: Bool) {
+    self.simulatorEvent()
+  }
+
+  public func agentDidLaunch(launchConfig: FBAgentLaunchConfiguration!, didStart agentProcess: FBProcessInfo!, stdOut: NSFileHandle!, stdErr: NSFileHandle!) {
+    self.simulatorEvent()
+  }
+
+  public func agentDidTerminate(agentProcess: FBProcessInfo!, expected: Bool) {
+    self.simulatorEvent()
+  }
+
+  public func applicationDidLaunch(launchConfig: FBApplicationLaunchConfiguration!, didStart applicationProcess: FBProcessInfo!, stdOut: NSFileHandle!, stdErr: NSFileHandle!) {
+    self.simulatorEvent()
+  }
+
+  public func applicationDidTerminate(applicationProcess: FBProcessInfo!, expected: Bool) {
+    self.simulatorEvent()
+  }
+
+  public func diagnosticInformationAvailable(name: String!, process: FBProcessInfo!, value: protocol<NSCoding, NSCopying>!) {
+    self.simulatorEvent()
+  }
+
+  public func didChangeState(state: FBSimulatorState) {
+    self.simulatorEvent()
+  }
+
+  public func terminationHandleAvailable(terminationHandle: FBTerminationHandleProtocol!) {
+    self.simulatorEvent()
+  }
+
+  public func report(eventName: EventName, _ eventType: EventType, _ subject: EventReporterSubject) {
+    do {
+      self.writer.write(
+        try self.json.serializeToString([
+          "simulator" : self.simulator.jsonSerializableRepresentation(),
+          "event_name" : eventName.rawValue,
+          "event_type" : eventType.rawValue,
+          "subject" : subject.jsonSerializableRepresentation()
+        ])
+      )
+    } catch {
```

Comment:
I was thinking we `abort` instead, it's programmer error if an event is passed that is un-seralizable.